.github/workflows/phpcs.yml

```
@@ -0,0 +1,42 @@
+name: PHPCS
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  phpcs:
+    name: WPCS
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Set PHP version
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '7.4'
+          coverage: none
+          tools: composer:v2
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v35
+        with:
+          files: |
+            **/*.php
+      
+      - name: Install dependencies
+        if: ${{ steps.changed-files.outputs.any_changed == 'true' }}
+        run: composer install
```

Mentioned this in a comment but I think we could do something like:
```suggestion
        run: |
          composer require --dev sirbrillig/phpcs-changed -n
          composer config --no-plugins allow-plugins.dealerdirect/phpcodesniffer-composer-installer true
          composer require --dev woocommerce/woocommerce-sniffs -n
```